rocks-db/src/lib.rs

```diff
@@ -538,6 +532,12 @@ impl Storage {
                     asset::AssetLeaf::merge_asset_leaf,
                 );
             },
+            asset::AssetLeafDeprecated::NAME => {
+                cf_options.set_merge_operator_associative(
+                    "merge_fn_merge_asset_leaf",
+                    asset::AssetLeaf::merge_asset_leaf,
```

Comment:
This merge function with real data will most likely fail on 
```
 if let Some(existing_val) = existing_val {
            match deserialize::<AssetLeaf>(existing_val) {
```
I'd like to see a test proving this wrong, like 
```
let deprecated = AssetLeafDepricated{...};
let serialised = serialise(deprecated);
let deserialised = deserialize::<AssetLeaf>(serialised);
```

Or did you mean to use AssetLeafDeprecated instead?

Reply:
Yes, I planed use AssetLeafDeprecated::merge_asset_leaf

Reply:
will fix the issue